thunder/examine/memory_caculation.py

```diff
@@ -118,6 +120,32 @@ def del_op_memory(bsym: BoundSymbol, tensor_to_memory_data: ProxyDict, name_to_a
     return memory_size
 
 
+@register_memory_calculate_function("clear_mutable_collection")
+def clear_mutable_collection_argument_memory(
+    bsym: BoundSymbol, tensor_to_memory_data: ProxyDict, name_to_alloc_memory: dict[str, int], is_argument: bool
+) -> int:
+    # Clearing the collection forces the interpreter to release references to its elements,
+    # even if the collection was an argument.
+    # So we cancel the n += 1 (see get_alloc_memory) for tensors contained in such a collection
+    if not is_argument:
+        return 0
+
+    collection_proxy = bsym.flat_proxy_args[0]
+    if not isinstance(collection_proxy.collection(), (MutableSequence, MutableMapping, MutableSet)):
```

Comment:
This check works because the updated `transforms.py` now sets as `bw_trace.args` the same kinds of collections that `ThunderFunction.backward` actually passes.

Reply:
The same check is in `thunder.executors.pythonex`.
https://github.com/Lightning-AI/lightning-thunder/blob/a89aa488aaa9220a03a82c982744529769f89b15/thunder/executors/pythonex.py#L252-L254